protos/flyteidl/core/tasks.proto

```diff
@@ -125,6 +125,12 @@ message TaskTemplate {
 
     // security_context encapsulates security attributes requested to run this task.
     SecurityContext security_context = 8;
+
+    // Metadata about the custom defined for this task. This is extensible to allow various plugins in the system
+    // to use as required.
+    // reserve the field numbers 1 through 15 for very frequently occurring message elements
+    map<string, string> config = 16;
```

Comment:
Should we mark the config inside the container as deprecated?

Reply:
sure, done @EngHabu